src/QsCompiler/TextProcessor/SyntaxBuilder.fs

```diff
@@ -182,16 +182,27 @@ let internal getStringContent interpolArg =
     let startDelimiter = pstring "\""
     let endDelimiter = notPrecededBySlash >>. startDelimiter
     let interpolatedString = 
-        let contentDelimiter = endDelimiter <|> (notPrecededBySlash >>. lCurly)
-        let nonInterpol = manyChars (notFollowedBy contentDelimiter >>. anyChar)
+        let interpolCharSnippet = manySatisfy (fun c-> c<> '\\' && c <> '"' && c <> '{')
+        let escapedChar = pstring "\\" >>. (anyOf "\\\"nrt{" |>> function // Also supports escapting '{'
```

Comment:
This pattern for escape handling is based on the documentation for [FParsec.CharParsers.stringsSepBy](http://www.quanttec.com/fparsec/reference/charparsers.html#members.stringsSepBy), and as mentioned in the other comment works for the `\n`, `\r\`, `\t`, and `\{` but somehow not for `\\` and `\"`. I'm going to keep tweaking it to see if I can figure out why, but I would've thought that the `| c -> string c` case would work for those characters. It already seems like that case is correctly handling `\{` so I wonder if there is something in one of the other parsers used here that is preventing it from working for `\\` and `\"`.

Reply:
Looking at the test, it looks like it's not that it's producing the wrong string, but rather the parsing itself is failing. I haven't spotted the issue yet either though.

Reply:
I see why it's failing to parse: `endDelimiter` parser looks for the string termination by parsing for a `"` that is not preceded by `\` but the test case is the literal string `"\"` which means the terminating quote IS preceded by slash. This means the current set of parsers would never be able to parse a string literal that ends in a slash, but at the same time the tests as written always look for the special character as the last character in the string literal. While I'm still not sure how to fix this, I don't think changing the test is the right thing here; a string literal should be allowed to end in an escaped slash.

Reply:
Oh, I get it now! Since the `escapedChar` parser correctly handles the escape characters including escaped quotes, the `endDelimiter` parser no longer needs to handle the slash, making it identical to the `startDelimiter`. I've collapsed them into just `delimiter` and switched to using the [FParsec.CharParsers.between](http://www.quanttec.com/fparsec/reference/primitives.html#members.between) to get a slightly more optimized version of `delimiter >>. content ,>> delimiter`. Now all the tests pass :)

Reply:
Well done figuring this out!